unittests/pythontests.py

```diff
@@ -80,3 +82,11 @@ def test_versions(self):
         with self.assertRaises(unittest.SkipTest):
             self.init(testdir, extra_args=['-Dpython=dir'])
         self.wipe()
+
+    @unittest.mock.patch('mesonbuild.mesonlib.Popen_safe', return_value=(None, '{"error": "hello!"}', None))
+    @unittest.mock.patch('mesonbuild.mlog.log')
+    def test_introspection_error(self, mlog_log, Popen_safe):
+        python = PythonExternalProgram('some-python', ['some-python'])
+
+        assert not python.sanity()
```

Comment:
Hmm, shouldn't this be a unittest assertFalse, not a regular assert?

Reply:
When tests are [run with `pytest`](https://github.com/mesonbuild/meson/search?q=pytest), either is acceptable.  In fact, this simplicity without loss of debugging information is [pytest's number one feature](https://docs.pytest.org/en/latest/#id1].

Reply:
Well yes, but if you don't have both pytest and pytest-xdist then the tests get run without pytest, just unittest, and that is a supported configuration...

Reply:
It's the same thing in this case, these functions just exist to provide a nicer error output. pytest rewrites the bytecode to replace the asserts with something similar in order to achieve that. It makes sense when you are comparing lists for eg, but as we are just checking a bool, is pretty much the same.

Reply:
If it's "pretty much the same thing" to you, it shouldn't be a problem to use unittest asserts for the sake of those odd people using unittest instead of pytest to run this testsuite :)